x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/tool/BaseRunAsSuperuserCommand.java

```diff
@@ -0,0 +1,308 @@
+                .get(settings) == false) throw new UserException(ExitCodes.CONFIG, "File realm must be enabled");
+        }
+        // Else it's either explicitly enabled, or not defined in the settings so it is implicitly enabled.
+    }
+
+    /**
+     * Checks that we can connect to the cluster and that the cluster health is not RED. It optionally handles
+     * retries as the file realm might not have reloaded the users file yet in order to authenticate our
+     * newly created file realm user.
+     */
+    private void checkClusterHealthWithRetries(Environment env, Terminal terminal, int retries, boolean force) throws Exception {
+        CommandLineHttpClient client = clientFunction.apply(env);
+        final URL clusterHealthUrl = createURL(new URL(client.getDefaultURL()), "_cluster/health", "?pretty");
+        final HttpResponse response;
+        try {
+            response = client.execute("GET", clusterHealthUrl, username, password, () -> null, this::responseBuilder);
+        } catch (Exception e) {
+            throw new UserException(ExitCodes.UNAVAILABLE, "Failed to determine the health of the cluster. ", e);
+        }
+        final int responseStatus = response.getHttpStatus();
+        if (responseStatus != HttpURLConnection.HTTP_OK) {
+            if (responseStatus == HttpURLConnection.HTTP_UNAUTHORIZED && retries > 0) {
+                terminal.println(
+                    Terminal.Verbosity.VERBOSE,
+                    "Unexpected http status [401] while attempting to determine cluster health. Will retry at most "
+                        + retries
+                        + " more times."
+                );
+                Thread.sleep(1000);
+                retries -= 1;
+                checkClusterHealthWithRetries(env, terminal, retries, force);
+            } else {
+                throw new UserException(
+                    ExitCodes.DATA_ERROR,
+                    "Failed to determine the health of the cluster. Unexpected http status [" + responseStatus + "]"
+                );
+            }
+        } else {
+            final String clusterStatus = Objects.toString(response.getResponseBody().get("status"), "");
+            if (clusterStatus.isEmpty()) {
+                throw new UserException(
+                    ExitCodes.DATA_ERROR,
+                    "Failed to determine the health of the cluster. Cluster health API did not return a status value."
+                );
+            } else if ("red".equalsIgnoreCase(clusterStatus) && force == false) {
+                terminal.errorPrintln("Failed to determine the health of the cluster. Cluster health is currently RED.");
+                terminal.errorPrintln("This means that some cluster data is unavailable and your cluster is not fully functional.");
+                terminal.errorPrintln("The cluster logs (https://www.elastic.co/guide/en/elasticsearch/reference/master/logging.html)" +
```

Comment:
I don't think we want to link to `/master/` here?

Linking to docs is tricky - historically we've avoided it, but I we're starting to do more of it.

We build a version specific URL in [SecurityStatusChangeListener](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityStatusChangeListener.java#L45-L47).  
Maybe we need to extract that to a "DocsLink" class?

Reply:
I'll build for the current version here and I'll open a PR/issue for extracting this to a class